google/cloud/storage/tools/run_testbench_utils.sh

```diff
@@ -75,18 +75,30 @@ start_testbench() {
   echo "Launching testbench emulator in the background"
   trap kill_testbench EXIT
 
-  # The tests typically run in a Docker container, where the ports are largely
-  # free; when using in manual tests, you can set EMULATOR_PORT.
-  local -r testbench_port=${TESTBENCH_PORT:-8000}
-
-  gunicorn --bind 0.0.0.0:${testbench_port} \
+  gunicorn --bind 0.0.0.0:0 \
       --worker-class gevent \
       --access-logfile - \
       --pythonpath "${PROJECT_ROOT}/google/cloud/storage/testbench" \
       testbench:application \
       >testbench.log 2>&1 </dev/null &
   TESTBENCH_PID=$!
 
+  local testbench_port="0"
+  for attempt in $(seq 1 8); do
+    if grep -q 'Listening at: http://0.0.0.0:' testbench.log; then
```

Comment:
It seems a shame to use three tools (grep, egrep, and sed) when just one (sed) will do.

For example, ...

```
  local testbench_port=
  readonly listening_at='Listening at: http://0.0.0.0:\([1-9][0-9]*\)'
  for attempt in $(seq 1 8); do
    testbench_port=$(sed -n "s,${listening_at},\1,p" testbench.log)
    [[ -n "${testbench_port}" ]] && break
    sleep 1
  done

  if [[ -z "${testbench_port}" ]]; then
    echo "Cannot find listening port for testbench." >&2
    exit 1
  fi
```